In [2]:
import pandas as pd

In [12]:
df=pd.read_csv('../Dataset/raw.csv')
df.head(5)

,gender,race_ethnicity,parental_level_of_education,lunch,test_preparation_course,math_score,reading_score,writing_score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93
3,male,group A,associate's degree,free/reduced,none,47,57,44
4,male,group C,some college,standard,none,76,78,75


<h2>2.2 Dataset information</h2>
gender : sex of students -> (Male/female)<br>
race/ethnicity : ethnicity of students -> (Group A, B,C, D,E)<br>
parental level of education : parents' final education ->(bachelor's degree,some college,master's degree,associate's degree,high school)<br>
lunch : having lunch before test (standard or free/reduced)<br>
test preparation course : complete or not complete before test<br>
math score<br>
reading score<br>
writing score<br>
<h2>3. Data Checks to perform</h2>
Check Missing values<br>
Check Duplicates<br>
Check data type<br>
Check the number of unique values of each column<br>
Check statistics of data set<br>
Check various categories present in the different categorical column<br>

<h2>3.1 Check Missing values</h2>


In [14]:
df.isna().sum()

gender                         0
race_ethnicity                 0
parental_level_of_education    0
lunch                          0
test_preparation_course        0
math_score                     0
reading_score                  0
writing_score                  0
dtype: int64

<h2> There are no missing Values in the data if found refer Missing Values Notebook

<H2>3.2 Check Duplicates

In [15]:
df.duplicated().sum()

np.int64(0)

<h2> There are no Duplicate  Values in the data if found refer Duplicate Values Notebook


<H2>3.3 Check Data types

In [17]:
# Check Null and Dtypes
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   gender                       1000 non-null   str  
 1   race_ethnicity               1000 non-null   str  
 2   parental_level_of_education  1000 non-null   str  
 3   lunch                        1000 non-null   str  
 4   test_preparation_course      1000 non-null   str  
 5   math_score                   1000 non-null   int64
 6   reading_score                1000 non-null   int64
 7   writing_score                1000 non-null   int64
dtypes: int64(3), str(5)
memory usage: 62.6 KB


<h2>3.4 Checking the number of unique values of each column

In [18]:
df.nunique()  #IMPPPPPPPPPPPPPPPPPPPPPPPP to USE THIS more high Values then mostly int they are !!!!!

gender                          2
race_ethnicity                  5
parental_level_of_education     6
lunch                           2
test_preparation_course         2
math_score                     81
reading_score                  72
writing_score                  77
dtype: int64

<h2>3.5 Check statistics of data set

In [19]:
df.describe()

,math_score,reading_score,writing_score
count,1000.00000,1000.000000,1000.000000
mean,66.08900,69.169000,68.054000
std,15.16308,14.600192,15.195657
min,0.00000,17.000000,10.000000
25%,57.00000,59.000000,57.750000
50%,66.00000,70.000000,69.000000
75%,77.00000,79.000000,79.000000
max,100.00000,100.000000,100.000000


<h4>write you insights </h4>
From above description of numerical data, all means are very close to each other - between 66 and 68.05;<br>
All standard deviations are also close - between 14.6 and 15.19;<br>
While there is a minimum score 0 for math, for writing minimum is much higher = 10 and for reading myet higher = 17<br>

<h2>3.7 Exploring Data

In [20]:
df.head()

,gender,race_ethnicity,parental_level_of_education,lunch,test_preparation_course,math_score,reading_score,writing_score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93
3,male,group A,associate's degree,free/reduced,none,47,57,44
4,male,group C,some college,standard,none,76,78,75


In [29]:
print("Categories in 'gender' variable:     ",end=" " )
print(df['gender'].unique())
print('\n')

print("Categories in 'race_ethnicity' variable:  ",end=" ")
print(df['race_ethnicity'].unique())
print('\n')


print("Categories in'parental level of education' variable:",end=" " )
print(df['parental_level_of_education'].unique())
print('\n')


print("Categories in 'lunch' variable:     ",end=" " )
print(df['lunch'].unique())
print('\n')



print("Categories in 'test preparation course' variable:     ",end=" " )
print(df['test_preparation_course'].unique())


Categories in 'gender' variable:      <StringArray>
['female', 'male']
Length: 2, dtype: str


Categories in 'race_ethnicity' variable:   <StringArray>
['group B', 'group C', 'group A', 'group D', 'group E']
Length: 5, dtype: str


Categories in'parental level of education' variable: <StringArray>
[ 'bachelor's degree',       'some college',    'master's degree',
 'associate's degree',        'high school',   'some high school']
Length: 6, dtype: str


Categories in 'lunch' variable:      <StringArray>
['standard', 'free/reduced']
Length: 2, dtype: str


Categories in 'test preparation course' variable:      <StringArray>
['none', 'completed']
Length: 2, dtype: str


In [35]:
# Numeric columns (int, float)
numeric_features = df.select_dtypes(include=['number']).columns.tolist()

# Categorical columns (strings or category dtype)
categorical_features = df.select_dtypes(include=['string', 'category']).columns.tolist()

print('We have {} numerical features : {}'.format(len(numeric_features), numeric_features))
print('We have {} categorical features : {}'.format(len(categorical_features), categorical_features))


We have 3 numerical features : ['math_score', 'reading_score', 'writing_score']
We have 5 categorical features : ['gender', 'race_ethnicity', 'parental_level_of_education', 'lunch', 'test_preparation_course']


In [36]:
df.head(2)

,gender,race_ethnicity,parental_level_of_education,lunch,test_preparation_course,math_score,reading_score,writing_score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88


<h2>3.8 Adding columns for "Total Score" and "Average"